### PR TITLE
feat: change get block height by ws protocol

### PR DESCRIPTION
**Pull request by joeylichang**
### Description

Change get block height by ws protocol, it is more lightweight.

### Rationale

N/A

### Example

N/A

### Changes

Notable changes: 

N/A

### CODE REVIEW

File: base/gnfd/gnfd.go [hunk 1]
```diff
@@ -7,12 +7,11 @@ import (
 	"sync"
 	"time"
 
-	"github.com/cosmos/cosmos-sdk/client/grpc/tmservice"
-
 	"github.com/bnb-chain/greenfield-storage-provider/base/types/gfsperrors"
 	"github.com/bnb-chain/greenfield-storage-provider/core/consensus"
 	"github.com/bnb-chain/greenfield-storage-provider/pkg/log"
 	chainClient "github.com/bnb-chain/greenfield/sdk/client"
+	chttp "github.com/cometbft/cometbft/rpc/client/http"
```
Comment (sysvm) [inline base/gnfd/gnfd.go:14]: pkg sorted

File: base/gnfd/gnfd.go [hunk 2]
```diff
@@ -49,10 +48,12 @@ type GnfdChainConfig struct {
 }
 
 type Gnfd struct {
-	client        *GreenfieldClient
-	backUpClients []*GreenfieldClient
-	stopCh        chan struct{}
-	mutex         sync.RWMutex
+	client          *GreenfieldClient
+	backUpClients   []*GreenfieldClient
```
Comment (sysvm) [inline base/gnfd/gnfd.go:52]: fix `backUpClients` to `backupClients `

File: base/gnfd/gnfd.go [hunk 3]
```diff
@@ -49,10 +48,12 @@ type GnfdChainConfig struct {
...
+	wsClient        *chttp.HTTP
+	backUpWsClients []*chttp.HTTP
```
Comment (sysvm) [inline base/gnfd/gnfd.go:54]: ditto

File: base/gnfd/gnfd_service.go [hunk 4]
```diff
@@ -20,12 +19,13 @@ import (
 
 // CurrentHeight the block height sub one as the stable height.
 func (g *Gnfd) CurrentHeight(ctx context.Context) (uint64, error) {
-	resp, err := g.getCurrentClient().chainClient.TmClient.GetLatestBlock(ctx, &tmservice.GetLatestBlockRequest{})
+	resp, err := g.getCurrentWsClient().ABCIInfo(ctx)
 	if err != nil {
-		log.CtxErrorw(ctx, "get latest block height failed", "node_addr", g.client.Provider, "error", err)
+		log.CtxErrorw(ctx, "get latest block height failed", "node_addr",
```
Comment (sysvm) [inline base/gnfd/gnfd_service.go:24]: fix to `failed to get latest block height`

**Review by will-2012 — APPROVED** [review]

**Pull request merged**